lib/razorpay/invoice.rb

```diff
@@ -5,52 +5,52 @@ module Razorpay
   # Invoice API allows you to create and
   # manage invoices with Razorpay
   class Invoice < Entity
-    def self.request
-      Razorpay::Request.new('invoices')
+    def initialize(auth_settings: {}, attributes: {})
+      super('invoices', auth_settings: auth_settings, attributes: attributes)
     end
 
-    def self.create(options)
+    def create(options)
       request.create options
     end
 
-    def self.fetch(id)
+    def fetch(id)
       request.fetch id
     end
 
-    def self.all(options = {})
+    def all(options = {})
       request.all options
     end
 
-    def self.edit(id, options = {})
+    def edit_with_id(id, options = {})
```

Comment:
We should document all the changes we have done to their methods in the README file itself. @ravi4k Please get this done, since we wont be going again and again in the source code to check the changes. And if we have this PR, then no one will know whether it was called edit or edit_with_id originally. Hence Please document

Reply:
Okay, I'll document the changes